### PR TITLE
Add transfer summary submit test

### PR DESCRIPTION
**Pull request by MancunianSam**
This is a test for when you submit the transfer summary form. It first checks to see if you're on the transfer confirmation page and then it will poll S3 every 10 seconds up to a timeout of 10 minutes to see if the expected export file is in S3.

### CODE REVIEW

File: application.conf [hunk 1]
```diff
@@ -1,5 +1,6 @@
 keycloak.user.admin.client="tdr-user-admin"
 tdr.base.url="http://localhost:9000"
-tdr.auth.url="http://localhost:8081"
-tdr.api.url="http://localhost:8080/graphql"
+tdr.auth.url="https://auth.tdr-integration.nationalarchives.gov.uk"
+tdr.api.url="https://api.tdr-integration.nationalarchives.gov.uk/graphql"
 browser="firefox"
+s3.bucket="tdr-consignment-export-intg"
```
Comment (TomJKing) [inline application.conf:6]: Missing line break ;)

File: application.staging.conf [hunk 2]
```diff
@@ -3,3 +3,5 @@ tdr.base.url="https://tdr-staging.nationalarchives.gov.uk"
 tdr.auth.url="https://auth.tdr-staging.nationalarchives.gov.uk"
 tdr.api.url="https://api.tdr-staging.nationalarchives.gov.uk/graphql"
 browser="firefox"
+s3.role="arn:aws:iam::"${INTG_AWS_ACCOUNT_NUMBER}":role/TDRJenkinsS3ExportRoleStaging"
```
Comment (TomJKing) [inline application.staging.conf:6]: Needs to be the staging aws account number.

File: application.staging.conf [hunk 3]
```diff
@@ -3,3 +3,5 @@ tdr.base.url="https://tdr-staging.nationalarchives.gov.uk"
...
+s3.bucket="tdr-consignment-export-staging"
```
Comment (TomJKing) [inline application.staging.conf:7]: Missing line break.

File: src/test/resources/features/TransferSummary.feature [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Feature: Transfer Summary Page
+  Scenario: Submitting the completed transfer summary page creates a completed export
+    Given A logged out user
```
Comment (TomJKing) [inline src/test/resources/features/TransferSummary.feature:3]: Is it worth adding some additional step or steps to describe the other prerequisites for an export, ie uploaded files, file checks completed?

File: src/test/resources/features/TransferSummary.feature [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    And an existing consignment for transferring body MOCK1
+    And the user is logged in on the Transfer Summary page
+    Then the user will be on a page with the title "Transfer Summary"
+    When the user selects yes to all transfer summary checks
+    And the user clicks the continue button
+    Then the user will be on a page with the title "Transfer Confirmation"
+    And the transfer export will be complete
```
Comment (TomJKing) [inline src/test/resources/features/TransferSummary.feature:10]: Missing line break.

File: src/test/scala/helpers/aws/AWSUtility.scala [hunk 6]
```diff
@@ -0,0 +1,51 @@
+package helpers.aws
+import java.time.LocalDateTime
```
Comment (TomJKing) [inline src/test/scala/helpers/aws/AWSUtility.scala:2]: Unnecessary import: `java.time.LocalDateTime`

File: src/test/scala/helpers/aws/AWSUtility.scala [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+import com.typesafe.config.ConfigFactory
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+import software.amazon.awssdk.http.apache.ApacheHttpClient
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sts.StsClient
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.{GetObjectRequest, ListObjectsRequest}
```
Comment (TomJKing) [inline src/test/scala/helpers/aws/AWSUtility.scala:11]: Unnecessary import: `ListObjectsRequest`

**Review by TomJKing — COMMENTED** [review]
Looks good. Some minor comments/suggestions

File: application.staging.conf (hunk 2 above)
Comment (MancunianSam) [reply application.staging.conf:6]: Good spot. I remember thinking, don't leave this as INTG without changing it and then clearly forgot.

File: src/test/resources/features/TransferSummary.feature (hunk 4 above)
Comment (MancunianSam) [reply src/test/resources/features/TransferSummary.feature:3]: Yeah, you're right, I'm not sure how it worked without an upload in the first place.
I've changed it so the file metadata checks are done and there is an actual file in the s3 bucket with the correct key and it seems to be working ok.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**